mlir/include/Quantum/IR/QuantumInterfaces.td

```diff
@@ -87,8 +99,29 @@ def QuantumGate : OpInterface<"QuantumGate"> {
         if (gate.getQubitOperands().size() < 1)
             return $_op->emitOpError("must have at least 1 qubit");
 
+        if (gate.getCtrlValueOperands().size() != gate.getCtrlQubitOperands().size())
+            return $_op->emitError() <<
+                "'" << $_op->getName() << "' op " <<
```

Comment:
These lines are not required, MLIR diagnostics already tell you which operation raised an error when calling `op->emitError()`.

Reply:
Done